test/test_pytato_transforms.py

```diff
@@ -0,0 +1,225 @@
+
+        if (
+            len(dag) == 1
+                and PrefixNamed("flux_container") in list(dag._data.values())[0].tags
+        ):
+            assert are_all_indexer_arrays_datawrappers(dag)
+            self.check_completed = True
+
+        return dag
+
+
+# {{{ test_resampling_indirections_are_fused_0
+
+def _compute_flux_0(dcoll, actx, u):
+    u_interior_tpair, = op.interior_trace_pairs(dcoll, u)
+    flux_on_interior_faces = u_interior_tpair.avg
+    flux_on_all_faces = op.project(
+        dcoll, FACE_RESTR_INTERIOR, FACE_RESTR_ALL, flux_on_interior_faces)
+
+    flux_on_all_faces = actx.tag(NameHint("flux_container"), flux_on_all_faces)
+    return flux_on_all_faces
+
+
+def test_resampling_indirections_are_fused_0(ctx_factory):
+    cl_ctx = ctx_factory()
+    cq = cl.CommandQueue(cl_ctx)
+
+    ref_actx = PyOpenCLArrayContext(cq)
+    actx = FluxOptimizerActx(cq)
+
+    dim = 3
+    nel_1d = 4
+    mesh = generate_regular_rect_mesh(
+        a=(-0.5,)*dim,
+        b=(0.5,)*dim,
+        nelements_per_axis=(nel_1d,)*dim,
+        boundary_tag_to_face={"bdry": ["-x", "+x",
+                                       "-y", "+y",
+                                       "-z", "+z"]}
+    )
+    dcoll = grudge.make_discretization_collection(ref_actx, mesh, order=2)
+
+    x, _, _ = dcoll.nodes()
+    compiled_flux_0 = actx.compile(lambda ary: _compute_flux_0(dcoll, actx, ary))
+
+    ref_output = ref_actx.to_numpy(
+        _compute_flux_0(dcoll, ref_actx, ref_actx.thaw(x)))
+    output = actx.to_numpy(
+        compiled_flux_0(actx.thaw(x)))
+
+    np.testing.assert_allclose(ref_output[0], output[0])
+    assert actx.check_completed
+
+# }}}
+
+
+# {{{ test_resampling_indirections_are_fused_1
+
+def _compute_flux_1(dcoll, actx, u):
+    u_interior_tpair, = op.interior_trace_pairs(dcoll, u)
+    flux_on_interior_faces = u_interior_tpair.avg
+    flux_on_bdry = op.project(dcoll, "vol", BTAG_ALL, u)
+    flux_on_all_faces = (
+        op.project(dcoll,
+                   FACE_RESTR_INTERIOR,
+                   FACE_RESTR_ALL,
+                   flux_on_interior_faces)
+        + op.project(dcoll, BTAG_ALL, FACE_RESTR_ALL, flux_on_bdry)
+    )
+
+    result = op.inverse_mass(dcoll, op.face_mass(dcoll, flux_on_all_faces))
+
+    result = actx.tag(NameHint("flux_container"), result)
+    return result
+
+
+def test_resampling_indirections_are_fused_1(ctx_factory):
+    cl_ctx = ctx_factory()
+    cq = cl.CommandQueue(cl_ctx)
+
+    ref_actx = PyOpenCLArrayContext(cq)
+    actx = FluxOptimizerActx(cq)
+
+    dim = 3
+    nel_1d = 4
+    mesh = generate_regular_rect_mesh(
+        a=(-0.5,)*dim,
+        b=(0.5,)*dim,
+        nelements_per_axis=(nel_1d,)*dim,
+        boundary_tag_to_face={"bdry": ["-x", "+x",
+                                       "-y", "+y",
+                                       "-z", "+z"]}
+    )
+    dcoll = grudge.make_discretization_collection(ref_actx, mesh, order=2)
```

Comment:
@inducer: I noticed a subtle and sub-optimal behavior that is worth noting. If we are using PyOpenCLActx as a setup actx and PytatoActx as an operational actx, then the frozen geometric factors (such as normals) would be stored at per-DOF even for affine geometry mappings. Such use-cases will bump memory footprint.